kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -0,0 +1,307 @@
+    private static final Logger logger = LoggerFactory.getLogger(KafkaServerExtension.class);
+
+    private AtomicBoolean initialized = new AtomicBoolean();
+    private Consumer<String, byte[]> consumer;
+    private ListenerSupport deploymentService;
+    private ProcessService processService;
+    private ScheduledExecutorService pollService;
+    private ExecutorService notifyService;
+
+    private Map<String, SignalInfo> topic2Signal = new ConcurrentHashMap<>();
+
+    @Override
+    public boolean isInitialized() {
+        return initialized.get();
+    }
+
+    @Override
+    public boolean isActive() {
+        return !Boolean.getBoolean(KieServerConstants.KIE_KAFKA_SERVER_EXT_DISABLED);
+    }
+
+    @Override
+    public void init(KieServerImpl kieServer, KieServerRegistry registry) {
+        if (initialized.getAndSet(true)) {
+            return;
+        }
+        pollService = Executors.newSingleThreadScheduledExecutor();
+        notifyService = Executors.newCachedThreadPool();
+        consumer = getKafkaConsumer();
+        KieServerExtension jbpmExt = registry.getServerExtension(JbpmKieServerExtension.EXTENSION_NAME);
+        if (jbpmExt == null) {
+            throw new IllegalStateException("Cannot find extension " + JbpmKieServerExtension.EXTENSION_NAME);
+        }
+
+        for (Object service : jbpmExt.getServices()) {
+            if (deploymentService == null && DeploymentService.class.isAssignableFrom(service.getClass())) {
+                deploymentService = (ListenerSupport) service;
+            } else if (processService == null && ProcessService.class.isAssignableFrom(service.getClass())) {
+                processService = (ProcessService) service;
+            }
+            if (deploymentService != null && processService != null) {
+                break;
+            }
+        }
+        if (deploymentService == null) {
+            throw new IllegalStateException("Cannot find deployment service");
+        }
+        if (processService == null) {
+            throw new IllegalStateException("Cannot find process service");
+        }
+        deploymentService.addListener(this);
+    }
+
+    @Override
+    public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
+        pollService.shutdownNow();
+        pollService = null;
+        notifyService.shutdownNow();
+        notifyService = null;
+        if (deploymentService != null) {
+            deploymentService.removeListener(this);
+            deploymentService = null;
+        }
+        if (consumer != null) {
+            consumer.close();
+            consumer = null;
+        }
+        initialized.set(false);
+    }
+
+    protected Consumer<String, byte[]> getKafkaConsumer() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, System.getProperty(
+                KAFKA_EXTENSION_PREFIX + "boopstrap.servers", "localhost:9092"));
```

Comment:
If you are setting by default this extension as true
https://github.com/kiegroup/droolsjbpm-integration/pull/2288/files#diff-821631efbb543afdf97147bcc31174b3fa3f48fc3b97159a0536a484468d0c4cR82
(disable false)

this is not necessaraly true as this can be active without any kafka server in the environment. Have you checked ?

Reply:
do we really need to check that? The kafka broker might be down when we start the server. I think this refers as if we want this extension to be active or not, regardless the status of the kafka broker

Reply:
Problem was that kie server wont start when extension is enabled and kafka broker down. Consumer is now lazily initialized to preven that. Deployment with signals/messages will fail if attempted when kafka broker is down (so user might retry)